datafusion/core/tests/user_defined/user_defined_plan.rs

```diff
@@ -290,10 +291,14 @@ async fn topk_plan() -> Result<()> {
 fn make_topk_context() -> SessionContext {
     let config = SessionConfig::new().with_target_partitions(48);
     let runtime = Arc::new(RuntimeEnv::default());
-    let mut state = SessionState::new_with_config_rt(config, runtime)
+    let state = SessionStateBuilder::new()
+        .with_config(config)
+        .with_runtime_env(runtime)
+        .with_default_features()
         .with_query_planner(Arc::new(TopKQueryPlanner {}))
-        .add_optimizer_rule(Arc::new(TopKOptimizerRule {}));
-    state.add_analyzer_rule(Arc::new(MyAnalyzerRule {}));
+        .add_optimizer_rule(Arc::new(TopKOptimizerRule {}))
```

Comment:
maybe we could make these names consistent (rather than `add_optimizer_rule` it could be `with_optimizer_rule` -- a follow on PR would be totally fine too

Reply:
will do, easy change.